<!doctype html><html><head><meta charset=UTF-8><title>08_Identify_variations - By</title><link rel=stylesheet href=https://cdn.staticfile.org/font-awesome/4.7.0/css/font-awesome.min.css><link rel=stylesheet href=https://cdn.staticfile.org/prism/1.15.0/themes/prism.min.css><link rel=stylesheet href=https://cdn.staticfile.org/KaTeX/0.10.0-rc.1/katex.min.css><link href=./css/chunk-vendors.0425ac81.css rel=stylesheet></head><body><div><article id=webslides><section slide class="slide bg-black-blue aligncenter" image="https://source.unsplash.com/C1HhAQrbykQ/ .dark"><span class="background dark" style="background-image:url('https://source.unsplash.com/C1HhAQrbykQ/')"></span><div class=wrap wrap=true><h1 class="text-landing text-shadow">08_Identify_variations</h1><hr><p class=text-intro>By Yan Li</p><p class=text-intro>PhD in Bioinformatics, University of Liverpool</p></div></section><section slide class="slide bg-light aligncenter"><div class=wrap wrap=true><h2>Why look at variation</h2><hr><ul><li>allows us to study evolution<ul><li>phylogenetics</li><li>population genomics</li></ul></li><li>let us make genotype / phenotype associations<ul><li>genome wide association studies (GWAS)</li><li>human disease, agriculture, genetic engineering</li></ul></li></ul></div></section><section slide class="slide bg-light aligncenter"><div class=wrap wrap=true><h2>The personal gene test service</h2><hr><p><img src=./img/wegene_1.b453ac59.png alt=wegene_1></p></div></section><section slide class="slide bg-light aligncenter"><div class=wrap wrap=true><p><img src=./img/wegene_2.625e20e9.png alt=wegene_2></p></div></section><section slide class="slide bg-light aligncenter"><div class=wrap wrap=true><p><img src=./img/wegene_3.8dfcbb0c.png alt=wegene_3></p></div></section><section slide class="slide bg-light aligncenter"><div class=wrap wrap=true><p><img src=./img/wegene_4.285f0107.png alt=wegene_4></p></div></section><section slide class="slide bg-light aligncenter"><div class=wrap wrap=true><h2>What is variation</h2><hr><ul><li>Single Nucleotide Polymorphisms (<strong>SNPs</strong>)<ul><li>a genetic &quot;typo&quot; of one nucleotide (e.g. A &gt; G)</li></ul></li><li>INsertion / DELetions (<strong>INDELs</strong>)<ul><li>a string of one or more nucleotides that has been added/removed from a location in a genome (typically 1-100bp)</li></ul></li><li>Structural Variants (<strong>SVs</strong>)<ul><li>a region of DNA that has been inverted / translocated / duplicated (typically &gt;100bp)</li></ul></li><li>Mobile Genetic Elements (<strong>MGEs</strong>)<ul><li>insertion / replication of retrotransposons, transposons, integrons etc.</li></ul></li></ul></div></section><section slide class="slide bg-light aligncenter"><div class=wrap wrap=true><h2>Workflow - GATK best practice</h2><hr><p><img src=./img/gatk_best_practice.20332731.webp alt=gatk></p><div><p>refer to <a href="https://software.broadinstitute.org/gatk/best-practices/workflow?id=11145" target=_blank>GATK website <i class="fa fa-external-link"></i></a></p></div></div></section><section slide class="slide bg-light aligncenter"><div class=wrap wrap=true><h2>Shors reads allignment</h2><hr><ul><li><strong>bwa</strong> (Burrows-Wheeler Aligner)</li><li>Bowtie</li></ul></div></section><section slide class="slide bg-light aligncenter"><div class=wrap wrap=true><h2>Sam/Bam files</h2><hr><ul><li><strong>SAM</strong> file is a TAB-delimited, line-oriented text format, including a header section and an alignment section<ul><li>Header section: each line contains some metadata</li><li>Alignment section: each line contains the alignment of a read</li><li>The <a href=https://samtools.github.io/hts-specs/SAMtags.pdf target=_blank>SAM tag specification</a></li></ul></li><li><strong>BAM</strong> file is the compressed format of SAM file</li></ul></div></section><section slide class="slide bg-light aligncenter"><div class=wrap wrap=true><h2>An example of SAM file</h2><hr><pre class=language-textile><code class=language-textile><span class="token phrase">@HD     VN:1.6  SO:coordinate
@SQ     SN:NC_003197.2  LN:4857450
@SQ     SN:NC_003277.2  LN:93933
@RG     ID:foo  PL:illumina     SM:SRR1056117
SRR10561173.6336        2209    NC<span class="token inline"><span class="token punctuation">_</span><span class="token italic">003197.2     1       60      57H44M  =       549     649     AGAGATTACGTCGGGTTGCAAGAGATCTTGACAGGGGGAATTGG    .G...G...GAA.&lt;A.&lt;A.GGGAGAGA.&lt;..G...&lt;GAA.GGAG    SA:Z:NC</span><span class="token punctuation">_</span></span>003197.2,4857394,+,57M44S,60,0; MC:Z:101M       MD:Z:12T14A16   PG:Z:MarkDuplicates     RG:Z:foo        NM:i:2  AS:i:34 XS:i:0
SRR10561173.114060      163     NC<span class="token inline"><span class="token punctuation">_</span><span class="token italic">003197.2     1       60      45S56M  =       178     275     GAAAAAAAACTAACAAAATAACGTGCTGTAATTTTTAAAATAATAAGAGATTACGTCTGGTTGCACGAGATCATGACAGGGGGAATTGGTTGAAAATAAAT   GGAA&lt;&lt;&lt;A.&lt;&lt;AGGGGAAGAGG.&lt;&lt;G.&lt;&lt;&lt;G.GGAA..&lt;GGA..&lt;...AA.&lt;AGA&lt;&lt;AG.GGGAG.&lt;.&lt;.&lt;.&lt;&lt;GG.GAGG...G.&lt;G.....&lt;&lt;.&lt;.GGG   SA:Z:NC</span><span class="token punctuation">_</span></span>003197.2,4857406,+,45M56S,60,0; MC:Z:98M        MD:Z:20A35      PG:Z:MarkDuplicates     RG:Z:foo        NM:i:1  AS:i:51 XS:i:0
</span></code></pre></div></section><section slide class="slide bg-light aligncenter"><div class=wrap wrap=true><h2>Call variants</h2><hr><ul><li>bcftools</li><li>FreeBayes</li><li><strong>GATK</strong><ul><li><strong>GATK</strong> v3.8 and v4.0 have some differences.</li><li>v4.0 merged the <code>picard</code> tools</li></ul></li><li>VarScan2</li></ul></div></section><section slide class="slide bg-light aligncenter"><div class=wrap wrap=true><h2>VCF files</h2><hr><p><img src=./img/vcf_file_format.1e11c474.jpg alt="vcf files"></p></div></section><section slide class="slide bg-light aligncenter"><div class=wrap wrap=true><h2>Annotate SNPs</h2><hr><ul><li><strong><code>snpeff</code></strong></li><li><code>Annovar</code></li><li><code>VAAST 2</code></li></ul></div></section><section slide class="slide bg-light aligncenter"><div class=wrap wrap=true><h2>Softwares</h2><hr><ul><li><a href=https://github.com/lh3/bwa target=_blank><code>bwa</code> <i class="fa fa-external-link"></i></a></li><li><a href=https://github.com/samtools/samtools target=_blank><code>samtools</code> <i class="fa fa-external-link"></i></a></li><li><a href=https://software.broadinstitute.org/gatk target=_blank><code>gatk</code> <i class="fa fa-external-link"></i></a></li><li><a href=https://snpeff.sourceforge.net/ target=_blank><code>snpeff</code> <i class="fa fa-external-link"></i></a></li></ul></div></section><section slide class="slide bg-light aligncenter"><div class=wrap wrap=true><h2>Reading</h2><hr><ul><li><a href=https://software.broadinstitute.org/gatk/best-practices target=_blank>gatk best practice <i class="fa fa-external-link"></i></a></li><li><a href=https://www.biotrainee.com/jmzeng/book/basic/filetype.html#sam target=_blank>Bioinformatics file format <i class="fa fa-external-link"></i></a></li></ul></div></section></article></div><script>window.pluginsOptions = {}



document.addEventListener('DOMContentLoaded', () => {
    let isPrintMode = false;
    if(~location.search.indexOf('print-pdf')){
        isPrintMode = true;
        WebSlides.registerPlugin('scroll', function(){});
    }
    const ws = new WebSlides({
        loop: false
    })
    window.wsInstance = ws;
    if(isPrintMode){
        ws.slides.forEach(s=>s.show())
    }
}, false)</script><script src=./js/chunk-vendors.e98a8274.js></script><script src=./js/08_Identify_variations.46c28b76.js></script></body></html>